astrolabe/runner.py

```diff
@@ -161,34 +157,35 @@ def run(self, persist_cluster=False, startup_time=1):
             driver_workload=self.spec.driverWorkload,
             startup_time=startup_time)
 
-        # Step-3: begin maintenance routine.
-        final_config = self.spec.maintenancePlan.final
-        cluster_config = final_config.clusterConfiguration
-        process_args = final_config.processArgs
+        for operation in self.spec.operations:
+            # Step-3: begin maintenance routine.
+            final_config = operation.setClusterConfiguration
```

Comment:
We should swap out everything inside this `for ...` with a single call to a utility function like `run_maintenance_operation`. That way there is an easy to find place to add maintenance operations.